app/src/test/java/org/mozilla/fenix/library/history/HistoryControllerTest.kt

```diff
@@ -143,4 +146,25 @@ class HistoryControllerTest {
         controller.handleDeleteSome(setOf(historyItem, newHistoryItem))
         assertEquals(itemsToDelete, setOf(historyItem, newHistoryItem))
     }
+
+    @Test
+    fun `openHistoryItem respects browsing mode`() {
+        val homeActivity = mockk<HomeActivity>()
+
+        every { homeActivity.browsingModeManager.mode } returns BrowsingMode.Private
+        DefaultHistoryController(
+            mockk(),
+            {
+                homeActivity.openToBrowserAndLoad(
```

Comment:
This test seems to just check that the lambda is called, so I think `onPressHistoryItemInNormalMode` already covers this. 

We might want to move `HistoryFragment#openItem` into the controller so you can test it, but that can be left for the future.

Reply:
I agree, the mechanism used for Bookmarks actually fits this. 

Regarding the redundant check, do you think it would be good to remove the `openToBrowserAndLoad` method?  Cause the original intent of the test is to prove that if a new tab is opened from history, then the browsing mode is not changed.

Reply:
We should remove the method, unless you feel like moving `openItem` to the controller (like Bookmarks). 

If you decide to move `openItem` then this is a good test to keep, otherwise you're really only testing the mock instead of the controller.

Reply:
Yes. Make sense. It does actually look like I'm testing the mock! Will remove this.

Reply:
Fixed. Sorry for resolving the conversation before actually pushing a commit. I realized later.